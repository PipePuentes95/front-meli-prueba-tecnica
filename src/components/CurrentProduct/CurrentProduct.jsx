import React from 'react';

import './CurrentProduct.scss'

const CurrentProduct = ({
  picture,
  condition,
  sold_quantity,
  title,
  price,
  description,
}) => {

  const formatAmount = (value) => {
    let format = Math.trunc(value);
    format = format.toString();
    return format.replace(/(\d)(?=(\d{3})+(?!\d))/g, '$1.');
  }

  const formatDecimals = (value) => {
    return value.toFixed(2).replace('0.', '');
  }

  return (
    <div className="current-product">
      <div className="cp-content">
        <div className="cp-img__wrapper">
          <img src={ picture } alt="img product" width="800" height="800" />
        </div>
        <div className="cp-summary">
          <div className="cp-condition-and-sold">
            <span>{ condition }</span> -  { sold_quantity > 0 && <span>{ sold_quantity } vendidos</span> } 
          </div>
          <h3 className='cp-product-name'>{ title }</h3>
          <p className='cp-price'>
            <span className='cp-price__amount'>$ { formatAmount(price?.amount) }</span>
            { price?.decimals !== 0 && <span className='cp-price__decimals'>{ formatDecimals(price?.decimals) }</span> }
          </p>
          <button className='btn-buy'>Comprar</button>
        </div>
      </div>

      <div className="cp-description">
        <h4 className='cp-description__title'>Descripci√≥n del producto</h4>
        <p className='cp-description__content'>{ description }</p>
      </div>
    </div>
  );
};

export default CurrentProduct;
