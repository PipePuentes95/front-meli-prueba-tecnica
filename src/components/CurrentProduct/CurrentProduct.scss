@import '../../styles/variables';

.current-product {
  @media(max-width: 764px) {
    width: 100%;
    padding: 16px;
  }

  width: 80%;

  margin: 0 auto;
  padding: 32px;

  background-color: $white;

  border-radius: 4px;
  overflow: hidden;

  .cp-content {
    display: flex;

    margin-bottom: 64px;
    
    @media(max-width: 769px) {
      flex-wrap: wrap;
    }

    .cp-img__wrapper {
      @media(min-width: 769px) {
        min-width: 600px;
        height: 600px;
      }

      @media(max-width: 768px) {
        margin-bottom: 16px;
      }

      width: 100%;
      height: 200px;
      
      img {
        width: 100%;
        height: 100%;

        object-fit: cover;
        object-position: center;
      }
    }

    .cp-summary {
      @media(min-width: 769px) {
        padding-left: 32px;
      }
    }

    .cp-condition-and-sold {
      margin-bottom: 16px;

      font-size: 14px;
      color: $grey-2;
      text-transform: capitalize;
    }

    .cp-product-name {
      margin-bottom: 32px;

      font-size: 24px;
      color: $grey-1;
      
      @media(max-width: 768px) {
        margin-bottom: 18px;

        font-size: 18px;
      }
    }

    .cp-price {
      display: flex;
      align-items: flex-start;

      margin-bottom: 32px;

      color: $grey-1;

      @media(max-width: 768px) {
        margin-bottom: 32px;
      }

      &__amount {
        margin-right: 4px;

        font-size: 46px;

        @media(max-width: 768px) {
          font-size: 32px;
        }
      }

      &__decimals {
        margin-top: 7px;

        font-size: 20px;

        @media(max-width: 768px) {
          font-size: 14px;
        }
      }
    }

    .btn-buy {
      width: 100%;

      padding: 14px;

      font-size: 20px;
      color: $white;

      background-color: $blue-1;

      border: none;
      border-radius: 4px;

      @media(max-width: 768px) {
        padding: 10px;
      }
    }
  }

  .cp-description {
    max-width: 700px;

    &__title {
      margin-bottom: 32px;

      font-size: 28px;
      font-weight: 500;
      color: $grey-1;

      @media(max-width: 768px) {
        margin-bottom: 20px;

        font-size: 28px;
      }
    }

    &__content {
      font-size: 16px;
      color: $grey-3;
    }
  }
}
