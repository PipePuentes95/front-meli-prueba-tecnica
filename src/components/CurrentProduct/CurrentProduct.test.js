import { render } from '@testing-library/react';
import CurrentProduct from './CurrentProduct';

const picture = "http://http2.mlstatic.com/D_950723-MLA47397900331_092021-I.jpg";
const  condition = "new";
const sold_quantity = 25;
const title = "Zapatillas Para Hombre Topper Squat Color Negro - Adulto 42 Ar";
const price = {
  "currency": "ARS",
  "amount": 9799,
  "decimals": 0
};
const description = "Topper se destaca en el mercado por ofrecer productos de calidad, confort y diseño que unen moda y deporte. Acerca tendencias y busca que las personas puedan verse y sentirse bien dentro y fuera del campo de juego.  \n\nSuela de goma \nLas zapatillas fabricadas con suela de goma tienen mayor resistencia al desgaste, son naturalmente impermeables y flexibles, y además extienden la estabilidad de la contracción.";

it('should take a snapshot', () => {
  const { asFragment } = render(
    <CurrentProduct 
      picture={ picture }
      condition={ condition }
      sold_quantity={ sold_quantity }
      title={ title }
      price={ price }
      description={ description } />
  );

  expect(asFragment(<CurrentProduct />)).toMatchSnapshot();
});