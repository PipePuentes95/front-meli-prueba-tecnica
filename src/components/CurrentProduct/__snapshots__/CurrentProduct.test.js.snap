// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="current-product"
  >
    <div
      class="cp-content"
    >
      <div
        class="cp-img__wrapper"
      >
        <img
          alt="img product"
          height="800"
          src="http://http2.mlstatic.com/D_950723-MLA47397900331_092021-I.jpg"
          width="800"
        />
      </div>
      <div
        class="cp-summary"
      >
        <div
          class="cp-condition-and-sold"
        >
          <span>
            new
          </span>
           -  
          <span>
            25 vendidos
          </span>
        </div>
        <h3
          class="cp-product-name"
        >
          Zapatillas Para Hombre Topper Squat Color Negro - Adulto 42 Ar
        </h3>
        <p
          class="cp-price"
        >
          <span
            class="cp-price__amount"
          >
            $ 9.799
          </span>
        </p>
        <button
          class="btn-buy"
        >
          Comprar
        </button>
      </div>
    </div>
    <div
      class="cp-description"
    >
      <h4
        class="cp-description__title"
      >
        Descripci칩n del producto
      </h4>
      <p
        class="cp-description__content"
      >
        Topper se destaca en el mercado por ofrecer productos de calidad, confort y dise침o que unen moda y deporte. Acerca tendencias y busca que las personas puedan verse y sentirse bien dentro y fuera del campo de juego.  

Suela de goma 
Las zapatillas fabricadas con suela de goma tienen mayor resistencia al desgaste, son naturalmente impermeables y flexibles, y adem치s extienden la estabilidad de la contracci칩n.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
